R/publish.R

```diff
@@ -8,23 +8,50 @@
 #' @param account Account name to publish to. Will default to any previously published
 #'  to account or any single account already associated with \code{server}.
 #' @param server Server to publish to (by default beta.rstudioconnect.com
-#' but any RStudio Connect server can be published to).
+#'  but any RStudio Connect server can be published to).
+#' @param render \code{TRUE} to render all formats prior to publishing (defaults to
+#'  \code{FALSE}, however this can be modified via the \code{bookdown.render_on_publish}
+#'  option). Note that this requires the use of a Makefile to provide the
+#'  implementaiton of rendering all formats.
 #'
 #' @export
 publish_book <- function(
-  name = NULL, account = NULL, server = 'beta.rstudioconnect.com'
+  name = NULL, account = NULL, server = NULL,
+  render = getOption("bookdown.render_on_publish", FALSE)
 ) {
 
   on.exit(opts$restore(), add = TRUE)
+
+  # render if requested (not we use the Makefile here rather than calling
+  # render_book because we need some mechanism that will have a-priori
+  # knowledge of all book formats)
+  if (isTRUE(render)) {
+    if (file.exists("Makefile")) {
+      result <- system2("make")
+      if (result != 0)
+        stop("Error ", result, " attempting to render book")
+    }
+    else
+      stop("Rendering before publishing requires a Makefile")
```

Comment:
Should we leave the Makefile in for now until we get
render_book('index.Rmd', 'all') working or should I remove the rendering
bit?

On Fri, Mar 18, 2016 at 1:06 PM, Yihui Xie <notifications@github.com> wrote:

> In R/publish.R
> <https://github.com/rstudio/bookdown/pull/53#discussion_r56689105>:
>
> >  ) {
> >
> >    on.exit(opts$restore(), add = TRUE)
> > +
> > +  # render if requested (not we use the Makefile here rather than calling
> > +  # render_book because we need some mechanism that will have a-priori
> > +  # knowledge of all book formats)
> > +  if (isTRUE(render)) {
> > +    if (file.exists("Makefile")) {
> > +      result <- system2("make")
> > +      if (result != 0)
> > +        stop("Error ", result, " attempting to render book")
> > +    }
> > +    else
> > +      stop("Rendering before publishing requires a Makefile")
>
> Oh right, that is an issue. In theory render_book('index.Rmd', 'all')
> should just work like render(), but I need to think more about it and try
> it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/bookdown/pull/53/files/1314b8f736c3be7b0e9df082a1f5a25eba16598d#r56689105>
>